addons/core/functions/fnc_requestRadios.sqf

```diff
@@ -33,62 +33,124 @@ GVAR(VehicleConfigCacheNamespace) setVariable ["TFAR_fnc_requestRadios_lastExec"
-                        private _source = _TF_SettingsToCopy select _copyIndex;
-                        private _variableName = format["%1_local", _source];
-                        private _localSettings = TFAR_RadioSettingsNamespace getVariable _variableName;
-                        if !(isNil "_variableName") then {
-                            [_destination, _localSettings, true] call TFAR_fnc_setSwSettings;
+[
+    "TFAR_RadioRequestResponseEvent",
+    {
+        if !(params [["_response", [], [[]]]]) exitWith {
+            diag_log ["TFAR_ReceiveRadioRequestResponse",_response];
+            hintC _response;
+            call TFAR_fnc_hideHint;
+            ["TFAR_RadioRequestResponseEvent", _thisId] call CBA_fnc_removeEventHandler;
+            [[15,"radioRequest",round ((diag_tickTime-TFAR_beta_RadioRequestStart)*1000)]] call TFAR_fnc_betaTracker;//#TODO remove on release
+        };
+        _thisArgs params ["_radiosToReplace", "_TF_SettingsToCopy", "_requestedUnit"];
+
+        diag_log ["TFAR_ReceiveRadioRequestResponse",_response];
+
+        {
+            if ((_x call TFAR_fnc_isRadio) || {_x == "ItemRadio"}) then {
+                private _index = _radiosToReplace find _x;
+                If (_index >= 0) then {
+                    private _newRadioClass = _response deleteAt _index;
+                    If (_newRadioClass isEqualTo "") then {
+                        _requestedUnit unassignItem _x;
+                        _requestedUnit removeItem _x;
+                    } else {
+                        _requestedUnit linkItem _newRadioClass;
+                    };
+
+                    _radiosToReplace deleteAt _index;
+                    private _settings = _TF_SettingsToCopy deleteAt _index;
+                    if ([_newRadioClass, _settings] call TFAR_fnc_isSameRadio) then {
+                        private _localSettings = TFAR_RadioSettingsNamespace getVariable (format["%1_local", _settings]);
+                        if !(isNil "_localSettings") then {
+                            [_newRadioClass, _localSettings, true] call TFAR_fnc_setSwSettings;
                         };
                     };
+
+                    [_newRadioClass, getPlayerUID player, true] call TFAR_fnc_setRadioOwner;
                 };
-                (_copyIndex + 1)
             };
-            private _linkFirstItem = TFAR_RadioReqLinkFirstItem;
-            if (_linkFirstItem) then {
-                TFAR_RadioReqLinkFirstItem = false;
-                TFAR_currentUnit linkItem (_response select 0);
-                _copyIndex = [_settingsCount, _copyIndex, (_response select 0),_TF_SettingsToCopy] call _fnc_CopySettings;
-                [(_response select 0), getPlayerUID player, true] call TFAR_fnc_setRadioOwner;
+            nil
+        } count (assignedItems _requestedUnit);
+
+        private _allItems = ((getItemCargo (uniformContainer TFAR_currentUnit)) select 0);
```

Comment:
I think it's better to just store the array of classnames sent to the server and use that. Instead of rescanning.

Reply:
If you do that you can ignore the _radiosToReplace find.
The server code should guarantee that the result is in the same Order.
And server should return number/string.
Number client will do oldClassname_Number
string if the server changed the radio (tf_ to TFAR_)
Empty string means just remove. addItem "" won't do anything so you don't really need to add special handling for that